### PR TITLE
DESCW-534 CleanBC patterns + BCgov base theme reusability

### PR DESCRIPTION
**Pull request by codewisenate**
Many patterns + BCgov base theme reuseability fixes (mostly around navigation) + DOM element selector checking for admin. Fixed bug in ribbon style navigation for offscreen flyout submenus

### CODE REVIEW

File: parts/header.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
-<!-- wp:group {"style":{"elements":{"link":{"color":{"text":"var:preset|color|white"}}},"spacing":{"padding":{"top":"1rem","right":"2rem","bottom":"0rem","left":"2rem"}}},"borderColor":"primary-brand-alt","backgroundColor":"primary-brand","textColor":"white","className":"bcgov-header-group bcgov-site-header","layout":{"inherit":true}} -->
-<div class="wp-block-group bcgov-header-group bcgov-site-header has-border-color has-primary-brand-alt-border-color has-white-color has-primary-brand-background-color has-text-color has-background has-link-color" style="padding-top:1rem;padding-right:2rem;padding-bottom:0rem;padding-left:2rem"><!-- wp:columns {"verticalAlignment":"center","align":"wide","style":{"spacing":{"padding":{"top":"0.2rem","bottom":"1rem"}}}} -->
+<!-- wp:group {"style":{"elements":{"link":{"color":{"text":"var:preset|color|white"}}},"spacing":{"padding":{"top":"0rem","right":"2rem","bottom":"0rem","left":"2rem"}}},"borderColor":"primary-brand-alt","backgroundColor":"primary-brand","textColor":"white","className":"bcgov-header-group bcgov-site-header","layout":{"inherit":true}} -->
```
Comment (ShawnTurple) [inline parts/header.html:1]: usually when there is a zero value, units are not required, linters often complain about this.

File: src/scripts/public/ribbon-navigation.js [hunk 2]
```diff
@@ -9,42 +9,51 @@ const domReady = () => {
 		 * Check for menu items being out of viewport and class is needed.
 		 */
 		function doBoundsCheck(t) {
```
Comment (ShawnTurple) [inline src/scripts/public/ribbon-navigation.js:11]: never a big fan of one letter variables, doesn't tell the developer very much about the variable.

**Review by ShawnTurple — APPROVED** [review]

File: parts/header.html (hunk 1 above)
Comment (codewisenate) [reply parts/header.html:1]: Carry over from the WP block editor. Adding a 0 to the padding/margin brings the unit value with it.

File: src/scripts/public/ribbon-navigation.js (hunk 2 above)
Comment (codewisenate) [reply src/scripts/public/ribbon-navigation.js:11]: I'll improve this next PR.

**Pull request merged**